src/lambda_functions/get_prices_lambda/scraper.py

```
@@ -365,398 +383,411 @@
+            self.performance_tracker.record_cache_miss(url)
+            
+            with self._measure_request(url):
+                response = self.session.get(url, timeout=self.config.get('timeout', 30))
+                response.raise_for_status()
+                data = self._parse_response(response.text, source)
+                self.cache_manager.save_to_cache(url, data)
+                return data
+        except requests.exceptions.RequestException as e:
+            error = HTTPError(f"Request failed: {str(e)}", getattr(e.response, 'status_code', None), url)
+            self.error_handler.handle_error(error)
+            raise error
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)